_posts/2023-01-17-Timebox-Your-Software-Work.md

```diff
@@ -0,0 +1,106 @@
+
+{% capture col1content %}
+
+```markdown
+# Goals
+- Finish writing paper
+
+
+
+
+```
+
+_Vague and possibly large task_
+
+{% endcapture %}
+{% capture col2content %}
+
+```markdown
+# Goals
+- Finish writing paper
+  - Create paper outline
+  - Finish writing introduction
+  - Check for dead hyperlinks
+  - Request internal review
+```
+
+_Modular and more understandable tasks_
+{% endcapture %}
+
+{% include two-col.html col1=col1content col2=col2content %}
+
+Breaking down large amounts of work as smaller chunks within our code helps to ensure long-term maintainability and understandability. Similarly, keeping our tasks small can help ensure our goals are achievable and understandable (to ourselves or others). Without this modularity, tasks can be impossible to achieve (subjective in nature) or very difficult to understand. Stated differently, taking many small steps can lead to a big change in an organized, oftentimes less exhausting way ([related graphic](https://www.fosslien.com/liz-fosslien-the-power-of-small-steps)).
+
+## Version Control and Timeboxing
+
+```markdown
+# Repo Issues
+- "Prevent foo warning" - 20 minutes
+- "Remove bar feature" - 20 minutes
+- "Address baz error" - 20 minutes
+
+```
+
+_List of example version control repository issues with associated time duration._
+{:.center}
+
+The parallels between the time we give a task and related code can work towards your benefit. For example, [Github Issues](https://github.com/features/issues) can be created to outline a timeboxed task which relates to a distinct chunk of code to be created, updated, or fixed. Once development tasks have been outlined as issues, a developer can use timeboxing to help organize how much time to allocate on each issue.
+
+Using Github Issues in this way provides a way to observe task progress associated with one or many repositories. It also increases collaborative opportunities for task sizing and description. For example, if a task looks too large to complete in a reasonable amount of time, developers may work together to break the task down into smaller modules of work.
+
+## Be Kind to Yourself: Take Breaks
+
+While timeboxing is often a conversation about how to be more productive, it's also worth remembering: __take breaks to be kind to yourself and more effective__. Some studies and thought leadership have the shown that taking breaks may be necessary to avoid performance decreases and impacts to your health. There's also some indication that taking breaks may lead to better work. See below for just a few examples:
```

Comment:
```suggestion
While timeboxing is often a conversation about how to be more productive, it's also worth remembering: __take breaks to be kind to yourself and more effective__. Some studies and thought leadership have shown that taking breaks may be necessary to avoid performance decreases and impacts to your health. There's also some indication that taking breaks may lead to better work. See below for just a few examples:
```